extensions/amp-crypto-polyfill/0.1/amp-crypto-polyfill.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as lib from '../../../third_party/closure-library/sha384-generated';
+
+// Register doc-service factory.
+AMP.registerServiceForDoc(
```

Comment:
Should we just do window-singleton service instead? It really always window-scoped, right? Just as WebCrypto itself...